enginetest/insert_queries.go

```diff
@@ -907,3 +909,136 @@ var InsertErrorScripts = []ScriptTest{
 		ExpectedErr: parse.ErrInvalidAutoIncCols,
 	},
 }
+
+var InsertIgnoreScripts = []ScriptTest{
+	{
+		Name: "Try INSERT IGNORE with primary key, non null, and single row violations",
+		SetUpScript: []string{
+			"CREATE TABLE y (pk int primary key, c1 int NOT NULL);",
+			"INSERT IGNORE INTO y VALUES (1, 1), (1,2), (2, 2), (3, 3)",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "SELECT * FROM y",
+				Expected: []sql.Row{
+					{1, 1}, {2, 2}, {3, 3},
+				},
+			},
+			{
+				Query: "INSERT IGNORE INTO y VALUES (1, 2), (4,4)",
+				Expected: []sql.Row{
+					{sql.OkResult{RowsAffected: 1}},
+				},
+				ExpectedWarning: mysql.ERDupEntry,
+			},
+			{
+				Query: "INSERT IGNORE INTO y VALUES (5, NULL)",
+				Expected: []sql.Row{
+					{sql.OkResult{RowsAffected: 1}},
+				},
+				ExpectedWarning: mysql.ERBadNullError,
+			},
+			{
+				Query: "INSERT IGNORE INTO y SELECT * FROM y WHERE pk=(SELECT pk FROM y WHERE pk > 1);",
+				Expected: []sql.Row{
+					{sql.OkResult{RowsAffected: 0}},
+				},
+				ExpectedWarning: mysql.ERSubqueryNo1Row,
+			},
```

Comment:
Throw in a case with a duplicate primary key but different `c1` value, so like `(4, 8)` or such.